### PR TITLE
Add support workspace_vcs resource

### LINKED ISSUE #41: Support VCS workspace type
With a quick looks, it seems we can refactor the existing `terrakube_worksapce_cli` resource to support VCS type?

### PR DESCRIPTION
**Pull request by alfespa17**
Fix #41

### CODE REVIEW

File: internal/provider/workspace_cli_resource.go [hunk 1]
```diff
@@ -387,13 +387,13 @@ func (r *WorkspaceCliResource) Delete(ctx context.Context, req resource.DeleteRe
 	workspaceCliRequest.Header.Add("Authorization", fmt.Sprintf("Bearer %s", r.token))
 	workspaceCliRequest.Header.Add("Content-Type", "application/vnd.api+json")
 	if err != nil {
-		resp.Diagnostics.AddError("Error creating team resource request", fmt.Sprintf("Error creating team resource request: %s", err))
+		resp.Diagnostics.AddError("Error creating vcs resource request", fmt.Sprintf("Error creating vcs resource request: %s", err))
```
Comment (stanleyz) [inline internal/provider/workspace_cli_resource.go:390]: vcs? cli, I presume

Comment (alfespa17) [reply internal/provider/workspace_cli_resource.go:390]: This is just a typo error in the workspace_cli_resource

**Comment by stanleyz** [comment]
I was wondering whether the existing `workspace_cli` can be refactored to `workspace` to support both cli and vcs type and potentially the API-driven workflow, but I am happy enough with this new resource also.

**Comment by alfespa17** [comment]
> I was wondering whether the existing `workspace_cli` can be refactored to `workspace` to support both cli and vcs type and potentially the API-driven workflow, but I am happy enough with this new resource also.

Workspace cli is a special case inside terrakube, I think it is better to have a different resource

**Comment by stanleyz** [comment]
Just tested this from my local, worked well. Do you mind add `Terraform Working Directory` as well? Thanks.

**Comment by alfespa17** [comment]
> Just tested this from my local, worked well. Do you mind add `Terraform Working Directory` as well? Thanks.

I will create another PR with the working directory, I also need to add the VCS connection so it can be used with private VCS

**Comment by stanleyz** [comment]
Thanks, and tags for workspaces. I might create two issues for each of these if you haven't. I might pick one for PR.

**Comment by alfespa17** [comment]
> Thanks, and tags for workspaces. I might create two issues for each of these if you haven't. I might pick one for PR.

If you want to help with the tags all help is welcome

**Pull request merged**